### PR TITLE
Add Octavia tests

### PR DESCRIPTION
**Pull request by fnordahl**

### CODE REVIEW

**Comment by thedac** [comment]
I merged the neutron-openvswitch tests which appeared to also be a part of this PR. A rebase will be necessary.

**Comment by thedac** [comment]
Looks good so far. I would only request the glance image creation has feature parity with https://github.com/juju/charm-helpers/blob/master/charmhelpers/contrib/openstack/amulet/utils.py#L683

**Comment by fnordahl** [comment]
> Looks good so far. I would only request the glance image creation has feature parity with https://github.com/juju/charm-helpers/blob/master/charmhelpers/contrib/openstack/amulet/utils.py#L683

Thanks for pointing that out @thedac , I will take a look and add missing functionality.  Need to teach it to consume artifacts from local object storage when environment variables present too.

File: zaza/charm_tests/octavia/setup.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import zaza.charm_tests.test_utils
+import zaza.charm_tests.glance.setup as glance_setup
+
+
+def add_amphora_image():
+    """Add Octavia ``amphora`` test image to glance."""
+    image_name = 'amphora-x64-haproxy'
+    glance_setup.add_image(
+        'http://tarballs.openstack.org/octavia/test-images/'
+        'test-only-amphora-x64-haproxy-ubuntu-xenial.qcow2',
```
Comment (thedac) [inline zaza/charm_tests/octavia/setup.py]: We will be able to publish a local image in object storage in serverstack we can use here as a default. But let's make this configurable.

File: zaza/charm_tests/octavia/tests.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                'loadbalancer': {
+                    'description': 'Created by Zaza',
+                    'admin_state_up': True,
+                    'vip_subnet_id': subnet_id,
+                    'name': 'zaza-lb-0',
+                }})
+        lb_id = result['loadbalancer']['id']
+
+        @tenacity.retry(wait=tenacity.wait_fixed(1),
+                        reraise=True, stop=tenacity.stop_after_delay(900))
```
Comment (thedac) [inline zaza/charm_tests/octavia/tests.py]: We might consider an exponential back off so as not to add to the delay.

**Pull request merged**